### ISSUE TITLE
REST API repository creation and retrieval fail

### ISSUE THREAD

**Issue by gregid** [eclipse/rdf4j#1756]
Related to #1389 and #1541
I've finally decided to update to the version supporting repository creation and retrieval (3.0.3) but I wasn't able to successfully execute these operations. Following https://rdf4j.org/documentation/rest-api/#repository-creation-and-reconfiguration and assuming it is a typo and the `PUT` method should be to: `<RDF4J_URL>/repositories` instead I tried:
```
PUT http://localhost:8787/rdf4j-server/repositories HTTP/1.1
Content-Type:  text/turtle

@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
@prefix rep: <http://www.openrdf.org/config/repository#>.
@prefix sr: <http://www.openrdf.org/config/repository/sail#>.
@prefix sail: <http://www.openrdf.org/config/sail#>.
@prefix ms: <http://www.openrdf.org/config/sail/memory#>.

[] a rep:Repository ;
   rep:repositoryID "test" ;
   rdfs:label "test memory store" ;
   rep:repositoryImpl [
      rep:repositoryType "openrdf:SailRepository" ;
      sr:sailImpl [
         sail:sailType "openrdf:MemoryStore" ;
         ms:persist true ;
         ms:syncDelay 120
      ]
   ].
```
and got: `HTTP/1.1 500 Request method 'PUT' not supported`

Then tried getting the configuration from the one I created:
```
GET http://localhost:8787/rdf4j-server/repositories/northwind1/config HTTP/1.1

```
this time I got: `HTTP/1.1 404`

What is the current status of both of these methods?

**Comment by jeenbroekstra** [comment]
With respect to the `GET` request: #1541 (retrieval and modification of the repo config) is a new feature that will be released as part of release 3.1.0 - so it's not in 3.0.3, but it _is_ available in the 3.1.0 Milestone 1 build we published today. I realize we don't make that clear in the REST API documentation - apologies for that. I'll add a little marker to make that more obvious.

As for the `PUT` request: you're using an incorrect endpoint, it needs the repository ID of the new repository at the end, like so:

    PUT http://localhost:8787/rdf4j-server/repositories/test

Note that, in 3.0.x, this command can be used to either create a new repo, or to update the config of an existing one. However, following the implementation of #1541 in 3.1.0, this will be changed slightly: in 3.1 you can only use the PUT command to create a new repository. If you wish to modify an existing repo's config, you will have to use a POST on the `repositories/test/config` endpoint.

**Comment by jeenbroekstra** [comment]
I've update the REST API docs to make it more clear what's new in release 3.1.0.
